taxi/tests/test_views.py

```diff
@@ -0,0 +1,233 @@
+        )
+        self.client.force_login(self.user)
+
+    @override_settings(AUTH_PASSWORD_VALIDATORS=[])
+    def test_create_driver(self):
+        form_data = {
+            "username": "new_user",
+            "password1": "Password1",
+            "password2": "Password1",
+            "first_name": "new_first_name",
+            "last_name": "new_last_name",
+            "license_number": "HHH12345",
+        }
+        self.client.post(reverse("taxi:driver-create"), data=form_data)
+        new_user = get_user_model().objects.get(username=form_data["username"])
+        self.assertEqual(new_user.first_name, "new_first_name")
+        self.assertEqual(new_user.last_name, "new_last_name")
+        self.assertEqual(new_user.license_number, "HHH12345")
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "search_term,expected_results",
+    [
+        pytest.param(
+            "testdriver",
+            ["testdriver1",
+             "testdriver2"],
+            id="partial_match"
+        ),
+        pytest.param(
+            "",
+            ["testdriver1", "testdriver2", "driver3"],
+            id="empty_search"
+        ),
+        pytest.param("nonexistent", [], id="no_results"),
+    ],
+)
+def test_driver_search(client, search_term, expected_results):
+    user = get_user_model().objects.create_user(
+        username="testuser", password="testpass"
+    )
+    client.force_login(user)
+
+    _ = Driver.objects.create(
+        username="testdriver1",
+        password="testpass",
+        license_number="AAA11111"
+    )
+    _ = Driver.objects.create(
+        username="testdriver2",
+        password="testpass",
+        license_number="BBB22222"
+    )
+    _ = Driver.objects.create(
+        username="driver3",
+        password="testpass",
+        license_number="CCC33333"
+    )
+
+    response = client.get(
+        reverse("taxi:driver-list"),
+        {"username": search_term}
+    )
+    assert response.status_code == 200
+
+    response_content = response.content.decode()
+    for driver_username in expected_results:
+        assert driver_username in response_content
+    for driver in Driver.objects.exclude(username__in=expected_results):
+        assert driver.username not in response_content
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "search_term,expected_results",
+    [
+        pytest.param("ModelX", ["ModelX1", "ModelX2"], id="partial_match"),
+        pytest.param(
+            "",
+            ["ModelX1", "ModelX2", "ModelY1"],
+            id="empty_search"),
+        pytest.param("nonexistent", [], id="no_results"),
+    ],
+)
+def test_car_search(client, search_term, expected_results):
+    user = get_user_model().objects.create_user(
+        username="testuser", password="testpass"
+    )
+    client.force_login(user)
+
+    manufacturer = Manufacturer.objects.create(name="TestManufacturer")
+
+    _ = Car.objects.create(model="ModelX1", manufacturer=manufacturer)
```

Comment:
In the future, you could omit this syntax
```suggestion
    Car.objects.create(model="ModelX1", manufacturer=manufacturer)
```